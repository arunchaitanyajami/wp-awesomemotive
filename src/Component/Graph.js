export default () => {
  return (
    <div>Graph</div>
  )
}
