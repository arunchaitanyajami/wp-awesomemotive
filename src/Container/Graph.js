import { compose } from '@wordpress/compose';
import { withSelect, withDispatch } from '@wordpress/data';
import Graph from '../Component/Graph';

const DATA_STORE_NAME = 'awesomemotive/data';
export const mapSelectToProps = ( select ) => {
	const { getData } = select( DATA_STORE_NAME );
	return {
		tableData: getData(),
		isLoading: false
	};
};

export const mapDispatchToProps = ( dispatch ) => {
	const { reloadData } = dispatch( DATA_STORE_NAME );

	return {
		reloadData
	};
};

export default compose( [
	withSelect( mapSelectToProps ),
	withDispatch( mapDispatchToProps ),
] )( Graph );
