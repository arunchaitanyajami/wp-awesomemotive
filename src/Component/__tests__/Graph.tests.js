import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Graph from '../Graph';

const props = {
	tableData: {
		graph: {},
		table: {},
	},
	reloadData: jest.fn(),
};

describe( 'Graph Component', () => {
	it( 'Should Render the default component with empty props', async () => {
		const { container } = render( <Graph { ...props } /> );
		expect( screen.queryByText( 'table' ) ).not.toBeInTheDocument();
		expect( container ).toMatchSnapshot();
	} );

	it( 'Should Render the component with table containing data', async () => {
		props.tableData.graph = {
			0: { date: 1657929219, value: 31776 },
			1: { date: 1658015619, value: 38863 },
			2: { date: 1658102019, value: 13690 },
			3: { date: 1658188419, value: 33777 },
			4: { date: 1658274819, value: 18959 },
			5: { date: 1658361219, value: 34964 },
			8: { date: 1658447619, value: 8006 },
		};

		props.tableData.table.title = 'Test Title';

		const { container } = render( <Graph { ...props } /> );

		expect( container.querySelector( 'table' ) ).toBeVisible();
		expect( container.querySelector( 'table > thead' ) ).toBeVisible();
		expect( container.querySelector( 'table > thead > tr' ) ).toBeVisible();
		expect( container.querySelector( 'table > caption' ) ).toBeVisible();
		expect( container.querySelector( 'table > tbody' ) ).toBeVisible();

		expect(
			container.querySelector( 'table > caption' ).innerHTML
		).toEqual( props.tableData.table.title );
		expect(
			container.querySelectorAll( 'table >  tbody > tr' ).length
		).toEqual( 7 );
	} );
} );
