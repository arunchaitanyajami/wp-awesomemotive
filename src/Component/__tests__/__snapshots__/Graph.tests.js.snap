// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph Component Should Render the component with table containing data 1`] = `
<div>
  <div
    class="awesomemotive-container"
  >
    <div
      class="awesomemotive-graph-reload"
    >
      <div>
        <svg
          aria-hidden="true"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.1 4.8l-3-2.5V4c-4.4 0-8 3.6-8 8 0 3.7 2.5 6.9 6 7.7.3.1.6.1 1 .2l.2-1.5c-.4 0-.7-.1-1.1-.2l-.1.2v-.2c-2.6-.8-4.5-3.3-4.5-6.2 0-3.6 2.9-6.5 6.5-6.5v1.8l3-2.5zM20 11c-.2-1.4-.7-2.7-1.6-3.8l-1.2.8c.7.9 1.1 2 1.3 3.1L20 11zm-1.5 1.8c-.1.5-.2 1.1-.4 1.6s-.5 1-.8 1.5l1.2.9c.4-.5.8-1.1 1-1.8s.5-1.3.5-2l-1.5-.2zm-5.6 5.6l.2 1.5c1.4-.2 2.7-.7 3.8-1.6l-.9-1.1c-.9.7-2 1.1-3.1 1.2z"
          />
        </svg>
      </div>
    </div>
    <div
      class="awesomemotive-graph-container"
    >
      <table
        class="graph"
      >
        <caption>
          Test Title
        </caption>
        <thead>
          <tr>
            <th
              scope="col"
            >
              Page Views
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style="height: 64%;"
          >
            <th
              scope="row"
            >
              Sat Jul 16 2022
            </th>
            <td>
              <span>
                31776
              </span>
            </td>
          </tr>
          <tr
            style="height: 78%;"
          >
            <th
              scope="row"
            >
              Sun Jul 17 2022
            </th>
            <td>
              <span>
                38863
              </span>
            </td>
          </tr>
          <tr
            style="height: 27%;"
          >
            <th
              scope="row"
            >
              Mon Jul 18 2022
            </th>
            <td>
              <span>
                13690
              </span>
            </td>
          </tr>
          <tr
            style="height: 68%;"
          >
            <th
              scope="row"
            >
              Tue Jul 19 2022
            </th>
            <td>
              <span>
                33777
              </span>
            </td>
          </tr>
          <tr
            style="height: 38%;"
          >
            <th
              scope="row"
            >
              Wed Jul 20 2022
            </th>
            <td>
              <span>
                18959
              </span>
            </td>
          </tr>
          <tr
            style="height: 70%;"
          >
            <th
              scope="row"
            >
              Thu Jul 21 2022
            </th>
            <td>
              <span>
                34964
              </span>
            </td>
          </tr>
          <tr
            style="height: 16%;"
          >
            <th
              scope="row"
            >
              Fri Jul 22 2022
            </th>
            <td>
              <span>
                8006
              </span>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        style="text-align: center;"
      >
        Time Stamp
      </div>
    </div>
  </div>
</div>
`;

exports[`Graph Component Should Render the default component with empty props 1`] = `
<div>
  <div
    class="awesomemotive-container"
  >
    <div
      class="awesomemotive-graph-reload"
    >
      <div>
        <svg
          aria-hidden="true"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.1 4.8l-3-2.5V4c-4.4 0-8 3.6-8 8 0 3.7 2.5 6.9 6 7.7.3.1.6.1 1 .2l.2-1.5c-.4 0-.7-.1-1.1-.2l-.1.2v-.2c-2.6-.8-4.5-3.3-4.5-6.2 0-3.6 2.9-6.5 6.5-6.5v1.8l3-2.5zM20 11c-.2-1.4-.7-2.7-1.6-3.8l-1.2.8c.7.9 1.1 2 1.3 3.1L20 11zm-1.5 1.8c-.1.5-.2 1.1-.4 1.6s-.5 1-.8 1.5l1.2.9c.4-.5.8-1.1 1-1.8s.5-1.3.5-2l-1.5-.2zm-5.6 5.6l.2 1.5c1.4-.2 2.7-.7 3.8-1.6l-.9-1.1c-.9.7-2 1.1-3.1 1.2z"
          />
        </svg>
      </div>
    </div>
    <div
      class="awesomemotive-graph-container"
    >
      <div
        style="text-align: center;"
      >
        Time Stamp
      </div>
    </div>
  </div>
</div>
`;
